mypy/typeops.py

```diff
@@ -1069,6 +1069,8 @@ class Status(Enum):
 def try_contracting_literals_in_union(types: Sequence[Type]) -> list[ProperType]:
     """Contracts any literal types back into a sum type if possible.
 
+    Requires a flattened union and does not descend into children.
```

Comment:
Maybe it's better to add an `assert` or transparently handle this correctly? Both usages of this in mypy seem fine but I think preventing misuse is good.

Reply:
I'd rather not - this function is called twice in the whole codebase, one of them in a rather hot function (subtyping check). While failure to pass a flattened union may result in some false positives, it isn't dangerous and can't cause any errors down the road.